lib/cdo/contact_rollups.rb

```diff
@@ -326,6 +327,40 @@ def self.insert_from_dashboard_census_submissions
     log_completion(start)
   end
 
+  def self.update_geo_from_school_data
+    start = Time.now
+    log "Updating user geo data from school data"
+
+    # State for schools is stored in state abbreviation. We need to convert
+    # to state name, so do this row-by-row using existing Ruby code for that
+    # conversion.
+
+    sql = "
+    SELECT users.email, schools.city, schools.state, schools.zip
+    FROM users
+    INNER JOIN school_infos ON school_infos.id = users.school_info_id
+    INNER JOIN schools ON schools.id = school_infos.school_id"
+
+    dataset = DASHBOARD_REPORTING_DB_READER[sql]
+
+    dataset.each do |user_and_geo|
+      state_code = user_and_geo[:state]
+      # convert from state code to state name
+      state = get_us_state_from_abbr(state_code, true)
+      next unless state.presence
+      city = user_and_geo[:city]
```

Comment:
City from `schools` is all caps, whereas city information in `contact_rollups` is first letter capitalized. Possible to standardize?

Reply:
We could potentially force everything to all lower or all upper. Let's see if this becomes a pain point.